src/main/java/fi/aalto/cs/intellij/activities/RequiredPluginsCheckerActivity.java

```diff
@@ -0,0 +1,213 @@
+        availableIdeaPluginDescriptors
+            .stream()
+            .filter(
+                availableDescriptor -> availableDescriptor.getPluginId().equals(PluginId.getId(id))
+            )
+            .findFirst()
+            .ifPresent(missingOrDisabledIdeaPluginDescriptors::add);
+      });
+    }
+  }
+
+  /**
+   * Get the list of all the available in the main JetBrains plugin repository.
+   *
+   * <p>Note! The descriptors for the not installed plugins do not exist in IJ until now.
+   */
+  private void getAvailablePluginsFromMainRepo() {
+    try {
+      availableIdeaPluginDescriptors = RepositoryHelper.loadPlugins(new BgProgressIndicator());
+    } catch (IOException ex) {
+      logger.error("Could not retrieve plugins data from main repository.", ex);
+    }
+  }
+
+  /**
+   * Sorts required plugins into missing and disabled and shows respective notifications.
+   */
+  private void checkPluginsStatusAndNotify() {
+    List<IdeaPluginDescriptor> missingPluginDescriptors = new ArrayList<>();
+    List<IdeaPluginDescriptor> disabledPluginDescriptors = new ArrayList<>();
+
+    for (IdeaPluginDescriptor descriptor : missingOrDisabledIdeaPluginDescriptors) {
+      if (!PluginManager.isPluginInstalled(descriptor.getPluginId())) {
+        missingPluginDescriptors.add(descriptor);
+      } else if (PluginManager.isDisabled(descriptor.getPluginId().getIdString())) {
+        disabledPluginDescriptors.add(descriptor);
+      }
+    }
+
+    if (missingPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsInstallation(missingPluginDescriptors);
+    } else if (disabledPluginDescriptors.size() > 0) {
+      notifyAndSuggestPluginsEnabling(disabledPluginDescriptors);
+    }
+  }
+
+  /**
+   * Notify with an option to enable all the required plugins.
+   */
+  private void notifyAndSuggestPluginsEnabling(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
+        "A+",
+        "A+",
+        "Some plugins must be and enabled for the A+ plugin to work properly "
+            + getPluginsNamesString(descriptors) + ".",
+        NotificationType.WARNING);
+
+    notification.addAction(new NotificationAction(
+        "Enable the required plugin(s) (" + getPluginsNamesString(descriptors) + ").") {
+
+      /**
+       * Activate all the required plugins.
+       */
+      @Override
+      public void actionPerformed(@NotNull AnActionEvent e, @NotNull Notification notification) {
+        descriptors.forEach(descriptor -> Objects
+            .requireNonNull(PluginManager.getPlugin(descriptor.getPluginId())).setEnabled(true));
+        notification.expire();
+      }
+    });
+
+    Notifications.Bus.notify(notification);
+  }
+
+  /**
+   * Notify with an option to install all the required plugins and suggest restart.
+   */
+  private void notifyAndSuggestPluginsInstallation(List<IdeaPluginDescriptor> descriptors) {
+    Notification notification = new Notification(
```

Comment:
I'd consider extracting this to a separate method, e.g. "notifyAboutMissingPlugins.". You could also have a separate "notifyAbout"-method that then would be used from both the "notifyAboutMissingPlugins" and "notifyAboutDisabledPlugins".